string[] discrete_input_name
string[] coil_name
string[] input_register_name
string[] holding_register_name